### PR TITLE
feat(calling): Support Picture in picture mode (WPB-10652) - Part 1

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10652" title="WPB-10652" target="_blank">StoryWPB-10652</a>  [Android] calling PiP mode
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

The PR will put allow the app to go in Picture-in-Picture mode when the user puts the app in background

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

### Attachments (Optional)

https://github.com/user-attachments/assets/0768f71f-6d0f-41d2-960e-62bc40043818

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: default.json [hunk 1]
```diff
@@ -53,6 +55,7 @@
             "default_backend_title": "wire-staging",
             "encrypt_proteus_storage": true,
             "analytics_enabled": true,
+            "picture_in_picture_enabled": true,
```
Comment (MohamadJaara) [inline default.json]: request: Staging and beta are supposed to mimic prod features, can you please disable it there

File: default.json [hunk 2]
```diff
@@ -65,6 +68,7 @@
             "mls_support_enabled": false,
             "encrypt_proteus_storage": true,
             "analytics_enabled": true,
+            "picture_in_picture_enabled": true,
```
Comment (alexandreferris) [inline default.json]: then it needs to be disabled here on `beta` as well

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallActivity.kt [hunk 3]
```diff
@@ -119,3 +131,22 @@ fun getOngoingCallIntent(
     addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
     putExtra(EXTRA_CONVERSATION_ID, conversationId)
 }
+
+private const val ASPECT_RATIO_NUMERATOR = 2
+private const val ASPECT_RATIO_DENOMINATOR = 3
+
+fun OngoingCallActivity.enterPiPMode(conversationId: ConversationId, userId: UserId) {
+    appLogger.i("OngoingCallActivity: Entering Picture-in-Picture mode..")
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallActivity.kt]:
```suggestion
    appLogger.i("$TAG Entering Picture-in-Picture mode..")
```

File: app/src/main/kotlin/com/wire/android/ui/calling/CallActivity.kt [hunk 4]
```diff
@@ -20,14 +20,14 @@ package com.wire.android.ui.calling
...
+import androidx.activity.ComponentActivity
 import androidx.activity.viewModels
-import androidx.appcompat.app.AppCompatActivity
 import androidx.lifecycle.lifecycleScope
 import com.wire.android.ui.AppLockActivity
 import com.wire.kalium.logic.data.id.QualifiedIdMapperImpl
 import kotlinx.coroutines.launch
 
-abstract class CallActivity : AppCompatActivity() {
+abstract class CallActivity : ComponentActivity() {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/CallActivity.kt]: What was the reason to switch `CallActivity` from `AppCompatActivity` to `ComponentActivity`?

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt [hunk 5]
```diff
@@ -131,7 +135,16 @@ fun OngoingCallScreen(
         flipCamera = sharedCallingViewModel::flipCamera,
         setVideoPreview = sharedCallingViewModel::setVideoPreview,
         clearVideoPreview = sharedCallingViewModel::clearVideoPreview,
-        onCollapse = { activity.moveTaskToBack(true) },
+        onCollapse = {
+            if (BuildConfig.PICTURE_IN_PICTURE_ENABLED && isPiPAvailableOnThisDevice) {
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt]: As this verification logic is used in more places, can't we have it as a variable at the top as well?

Something like:
```
val shouldUsePiPMode = BuildConfig.PICTURE_IN_PICTURE_ENABLED && isPiPAvailableOnThisDevice

then use it as:
if (shouldUsePiPMode) { .. }
```

**Review by alexandreferris — COMMENTED** [review]
Great work! Just have a couple of nitpicks and questions

File: app/src/main/kotlin/com/wire/android/ui/calling/CallActivity.kt (hunk 4 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/CallActivity.kt:30]: It's a leftover I was trying using `onUserLeaveHint`

Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/calling/CallActivity.kt]: `onUserLeaveHint` is not supported in `AppCompatActivity`

**Review by yamilmedina — APPROVED** [review]
👏

**Review by alexandreferris — APPROVED** [review]

**Pull request merged**